### PR TITLE
Better handling of importing models so we can deal with renamed QUARK archives

### PR DESCRIPTION
**Pull request by linucks**

### CODE REVIEW

File: ample/util/tm_util.py [hunk 1]
```diff
@@ -503,8 +503,8 @@ def _mod_structures(self, model_aln, structure_aln, model_pdb, structure_pdb):
         # Alignment not always identical, let's aim for 90%
         identical = set(_model_data).intersection(set(_structure_data))
         if len(identical) / float(len(_model_data)) < 0.90:
-            msg = "Differing residues in model and structure. Affected PDBs %s - %s"
-            raise RuntimeError(msg % (model_name, structure_name))
+            msg = "Differing residues in model and structure. Affected PDBs %s - %s\n%s\n%s"
```
Comment (hlasimpk) [inline ample/util/tm_util.py:506]: `os.linesep` is probably cleaner than `\n`

File: ample/util/ample_util.py [hunk 2]
```diff
@@ -87,36 +89,55 @@ def extract_and_validate_models(amoptd, sequence=None, single=True, allsame=True
        only extract a file if the suffix is in the list
 
     """
+    
+    def is_tar_archive(filename):
```
Comment (hlasimpk) [inline ample/util/ample_util.py]: I think you might be able to use `tarfile.is_tarfile()` to do this

File: ample/util/shelxe.py [hunk 3]
```diff
@@ -12,9 +12,13 @@
 from ample.util import ample_util
 from ample.util import mtz_util
 
-mrbumpd = os.path.join(os.environ['CCP4'], "share", "mrbump", "include", "parsers")
-sys.path.insert(0,mrbumpd)
-import parse_shelxe
+
+try:
```
Comment (hlasimpk) [inline ample/util/shelxe.py:16]: Maybe mark the old mrbump import method for deprecation in the future

File: ample/util/tests/test_ample_util.py [hunk 4]
```diff
@@ -42,18 +42,42 @@ def test_extract_and_validate_models(self):
...
+        amoptd = {'models' : os.path.join(self.testfiles_dir, 'result.tar.bz2'),
+                  'models_dir' : temp_dir,
+                  'work_dir' : '.'}
         sequence = 'QPRRKLCILHRNPGRCYDKIPAFYYNQKKKQCERFDWSGCGGNSNRFKTIEECRRTCIG'
         models = ample_util.extract_and_validate_models(amoptd=amoptd, sequence=sequence, single=True)
         self.assertTrue(amoptd['quark_models'])
         self.assertEqual(len(models), 200)
         shutil.rmtree(temp_dir)
+        
+    def test_extract_and_validate_models_quark_zip(self):
```
Comment (hlasimpk) [inline ample/util/tests/test_ample_util.py:54]: This is probably not necessary but you could avoid adding new test files by simply taking the alldecoy.pdb file and tarring it/zipping it in certain ways before testing. This way you would be able to add tests for all the different types of tarring. Additionally you might want to add tests here to make sure that the function doesn't work with incorrect file types.

File: ample/util/tm_util.py (hunk 1 above)
Comment (linucks) [reply ample/util/tm_util.py:506]: True, but it's within a string, so I think passing os.linesep in and using another %s would be harder to read.

File: ample/util/ample_util.py (hunk 2 above)
Comment (linucks) [reply ample/util/ample_util.py]: Nice! There's a similar function in zipfile too so I'lll use that as well.

File: ample/util/tests/test_ample_util.py (hunk 4 above)
Comment (linucks) [reply ample/util/tests/test_ample_util.py:54]: Agreed, but one for a later commit.

**Pull request merged**